vllm/model_executor/models/registry.py

```diff
@@ -410,10 +415,83 @@ class _LazyRegisteredModel(_BaseRegisteredModel):
     module_name: str
     class_name: str
 
-    # Performed in another process to avoid initializing CUDA
+    @staticmethod
+    def _get_cache_dir() -> pathlib.Path:
+        return pathlib.Path(envs.VLLM_CACHE_ROOT).joinpath("modelinfos")
+
+    def _get_cache_filename(self) -> str:
+        cls_name = f"{self.module_name}-{self.class_name}".replace(".", "-")
+        return f"{cls_name}.json"
+
+    def _load_modelinfo_from_cache(self,
+                                   module_hash: str) -> _ModelInfo | None:
+        modelinfo_path = self._get_cache_dir().joinpath(
+            self._get_cache_filename())
+        if not modelinfo_path.exists():
+            return None
+
+        with open(modelinfo_path, encoding="utf-8") as file:
+            mi_dict = json.load(file)
+
+        if mi_dict["hash"] != module_hash:
+            return None
+
+        # file not changed, use cached _ModelInfo properties
+        return _ModelInfo(**mi_dict["modelinfo"])
+
+    def _save_modelinfo_to_cache(self, mi: _ModelInfo,
+                                 module_hash: str) -> None:
+        """save dictionary json file to cache"""
+        mi_dict = {}
+        for f in fields(mi):
+            mi_dict[f.name] = getattr(mi, f.name)
+        modelinfo_dict = {
+            "hash": module_hash,
+            "modelinfo": mi_dict,
+        }
+        modelinfo_path = self._get_cache_dir().joinpath(
+            self._get_cache_filename())
+        with open(modelinfo_path, "w", encoding="utf-8") as f:
+            json.dump(modelinfo_dict, f, indent=2)
+
+    @logtime(logger=logger)
     def inspect_model_cls(self) -> _ModelInfo:
-        return _run_in_subprocess(
-            lambda: _ModelInfo.from_model_cls(self.load_model_cls()))
+        from vllm.model_executor.model_loader.weight_utils import get_lock
+
+        model_path = pathlib.Path(__file__).parent.joinpath(
+            f"{self.module_name.split('.')[-1]}.py")
+
+        assert model_path.exists(), \
+            f"Model {self.module_name} expected to be on path {model_path}"
+        with open(model_path, "rb") as f:
+            module_hash = hashlib.md5(f.read()).hexdigest()
+
+        with get_lock(self._get_cache_filename(), self._get_cache_dir()):
+            try:
+                mi = self._load_modelinfo_from_cache(module_hash)
+                if mi is not None:
+                    logger.debug(("Loaded model info "
+                                  "for class %s.%s from cache"),
+                                 self.module_name, self.class_name)
+                    return mi
+            except Exception:
+                logger.exception(
```

Comment:
log the actual exception here (at least the name?)

Reply:
If there are specific exceptions that we expect catch them directly

Reply:
Did not know about exception, and yeah I guess we can always fall back to regular loading